Source/WebCore/platform/graphics/cocoa/IOSurface.mm

```diff
@@ -337,14 +334,14 @@ static IntSize computeMaximumSurfaceSize()
     return MachSendRight::adopt(IOSurfaceCreateMachPort(m_surface.get()));
 }
 
-RetainPtr<CGImageRef> IOSurface::createImage()
+RetainPtr<CGImageRef> IOSurface::createImage(CGContextRef context)
 {
-    return adoptCF(CGIOSurfaceContextCreateImage(ensurePlatformContext()));
+    return adoptCF(CGIOSurfaceContextCreateImage(context));
 }
 
-RetainPtr<CGImageRef> IOSurface::sinkIntoImage(std::unique_ptr<IOSurface> surface)
+RetainPtr<CGImageRef> IOSurface::sinkIntoImage(std::unique_ptr<IOSurface>, RetainPtr<CGContextRef> context)
 {
-    return adoptCF(CGIOSurfaceContextCreateImageReference(surface->ensurePlatformContext()));
+    return adoptCF(CGIOSurfaceContextCreateImageReference(context.get()));
```

Comment:
Can we use `IOSurfaceContextGetSurface` to verify that `context` does indeed wrap this IOSurface?

Reply:
done